### PR TITLE
Fix missing connection in some cases.

### PR DESCRIPTION
**Pull request by farmaazon**
### Pull Request Description
Introducing "labeled ports" and insertion points for expected arguments changes some Span-tree assertions we have in mind when implementing "finding by AST crumbs". This caused graph controller miss some connections in connection list - those were treated as "unconvertable" to span-tree crumbs.

### Important Notes
Applied hotfix, and left TODO to consider how to do it better.

### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Rust](https://github.com/luna/enso/blob/main/docs/style-guide/rust.md) style guide.
- [x] All code has automatic tests where possible.
- [x] All code has been profiled where possible.
- [x] All code has been manually tested in the IDE.
- [x] All code has been manually tested in the "debug/interface" scene.
- [x] All code has been manually tested by the PR owner against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).
- [ ] All code has been manually tested by at least one reviewer against our [test scenarios](https://docs.google.com/spreadsheets/d/1RatJDM_f9_3bvYhl3Bpq2d8SyKgtVdrV1RkGxPU17c8/edit?ts=5faa7049#gid=0).

### CODE REVIEW

File: src/rust/ide/lib/span-tree/src/generate.rs [hunk 1]
```diff
@@ -973,6 +973,41 @@ mod test {
...
+
+        // === Partial application chain - this argument ===
+
+        let ast = parser.parse_line("here.foo").unwrap();
+        let invocation_info = CalledMethodInfo {
+            parameters : vec![this_param.clone(), param1.clone(), param2.clone()]
+        };
+        let ctx      = MockContext::new_single(ast.id.unwrap(),invocation_info);
+        let mut tree = SpanTree::new(&ast,&ctx).unwrap() : SpanTree;
+        // assert!(false, "{:?}", tree);
```
Comment (wdanilo) [inline src/rust/ide/lib/span-tree/src/generate.rs:986]: rm

File: src/rust/ide/lib/span-tree/src/node.rs [hunk 2]
```diff
@@ -474,11 +474,19 @@ impl<'a,T:Payload> Ref<'a,T> {
...
             // therefore have different meaning!
-            let next = children.find_position(|ch| {
+            let next = self.node.children.iter().find_position(|ch| {
                 !ch.ast_crumbs.is_empty() && ast_crumbs.starts_with(&ch.ast_crumbs)
+            }).or_else(|| {
+                // We do the second round to cover case of "prefix-like" nodes with
+                // `InsertionPoint(ExpectedArgument(_))`. See also docs for
+                // `generate::generate_expected_argument`.
+                // TODO[ao]: This is kinda heuristic. Should be reconsidered in
+                //  https://github.com/enso-org/ide/issues/787
```
Comment (wdanilo) [inline src/rust/ide/lib/span-tree/src/node.rs:486]: This comment is not understandable for anyone but the author. Could you elaborate more why this is heuristic, what could be better ways to do it etc, please?

File: src/rust/ide/lib/span-tree/src/node.rs [hunk 3]
```diff
@@ -474,11 +474,19 @@ impl<'a,T:Payload> Ref<'a,T> {
...
-            // Please be advised, that the `ch.ast_crumhs` is not a field of Ref, but Child, and
+            // Please be advised, that the `ch.ast_crumbs` is not a field of Ref, but Child, and
             // therefore have different meaning!
-            let next = children.find_position(|ch| {
+            let next = self.node.children.iter().find_position(|ch| {
                 !ch.ast_crumbs.is_empty() && ast_crumbs.starts_with(&ch.ast_crumbs)
+            }).or_else(|| {
+                // We do the second round to cover case of "prefix-like" nodes with
+                // `InsertionPoint(ExpectedArgument(_))`. See also docs for
+                // `generate::generate_expected_argument`.
```
Comment (wdanilo) [inline src/rust/ide/lib/span-tree/src/node.rs:484]: What is a "second round"? I feel these docs here are very important but I also feel they are too short and not very understandable for someone not knowing this part of code really, really well.

**Review by wdanilo — APPROVED** [review]

File: src/rust/ide/lib/span-tree/src/node.rs (hunk 2 above)
Comment (farmaazon) [reply src/rust/ide/lib/span-tree/src/node.rs:486]: But there is explanation, just a line above?

File: src/rust/ide/lib/span-tree/src/node.rs (hunk 3 above)
Comment (farmaazon) [reply src/rust/ide/lib/span-tree/src/node.rs]: This is second round of finding node, as I thought can be deduced from the code. I'll extend the comment a little.

**Pull request merged**